File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        //brute force
        //computing difference for every pair of i and j (indices) and finding max among them   
        int ans = 0;

        for(int i = 0; i < prices.size(); i++){
            for(int j = i + 1; j < prices.size(); j++){
                ans = max(ans , prices[j]-prices[i]);
            }
        }
        return ans;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let mini = prices[0];
    let ans = 0;
    
    for(let el of prices){
        let cost = el - mini;
        ans = Math.max(ans,cost);
        mini = Math.min(el,mini);
    }

    return ans;
};